west.yml

```diff
@@ -1,52 +1,89 @@
-# The west manifest file for the nRF Connect SDK (NCS).
+# The west manifest file (west.yml) for the nRF Connect SDK (NCS).
 #
-# The per-installation west configuration file specifies the location of this
-# manifest file. The "path" option in the [manifest] section of .west/config
-# defines the folder that contains west.yml. The path is relative to the folder
-# that contains .west.
+# The per-workspace west configuration file, ncs/.west/config,
+# specifies the location of this manifest file like this:
 #
-# You can create your own manifest files and put them in any
-# repository you want, to create your own custom NCS installation.
-# For example, you could create a manifest file in your own
-# out-of-tree application directory, which would pull the nrf repository
-# in as an ordinary project.
+#     [manifest]
+#     path = nrf
 #
-# You can pass your manifest repositories to west init when creating a
-# new NCS installation. See the west documentation for more
-# information.
+# See the west documentation for more information:
+#
+# https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/west/index.html
 
 manifest:
-  defaults:
-    remote: ncs
+  # This west.yml requires west 0.7 or later, because the "import"
+  # feature used below was introduced then.
+  version: 0.7
```

Comment:
This should be minimum: `0.7.1` or we should merge this PR after next mergeup of Zephyr.

In windows, user's using west `0.7.0` will experience the following:
```
CMake Error at C:/Program Files/CMake/share/cmake-3.14/Modules/CheckCCompilerFlag.cmake:41 (set):
  Syntax error in cmake code at

    C:/Program Files/CMake/share/cmake-3.14/Modules/CheckCCompilerFlag.cmake:41

  when parsing string

    -fmacro-prefix-map=C:\tmp\test_ncs_pr\1942\ncs=WEST_TOPDIR

  Invalid character escape '\1'.
```

due to the issue:
https://github.com/zephyrproject-rtos/west/issues/374

So either wait for an upmerge, containing this fix: 
https://github.com/zephyrproject-rtos/zephyr/pull/22688

or bump the required version to `0.7.1`

Reply:
I don't think we need to wait until the next upmerge. In fact, I'd very much like to get this in *before* the next upmerge. Instead, let's require 0.7.x in the documentation here. @ru-fu do we need to state the minimum required version anywhere else?

Reply:
I don't actually think this particular line of west.yml needs to change, because the schema version is not the west version.

The current *manifest schema version* is 0.7. This is the west.yml schema supported by all west 0.7.x versions. 

Some versions (e.g. 0.7.0) have bugs which make us not want to use them. But the schema version for all of them is exactly the same.

This line is about the schema version, not the west version. So I think it is OK. The minimum west version can be set in nrf/requirements.txt.

Reply:
>  In fact, I'd very much like to get this in _before_ the next upmerge. Instead, let's require 0.7.x in the documentation here.

We may not be able to build the zephyr documentation without doing a mergeup. Let's see.

Reply:
> I don't actually think this particular line of west.yml needs to change, because the schema version is not the west version.

From NCS developer point of view, I completely agree, as this is not the meaning of this field.
But a customer who runs into the mentioned issue basically doesn't care how we solve it, he just gets annoyed if wasting time on something we could easily have solved.

Based on @carlescufi and @mbolivar comments, I created this:
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/285

So let get that change in, along with this.

Reply:
> But a customer who runs into the mentioned issue basically doesn't care how we solve it, he just gets annoyed if wasting time on something we could easily have solved.

Thanks for sending a fix, but I don't think we could fix this in west.yml at all, could we?

Reply:
> Thanks for sending a fix, but I don't think we could fix this in west.yml at all, could we?

Well, if we request version `0.7.1` in `west.yml` it would fail with a proper error message if user is using `0.7.0` and with a cryptic message if `0.6.x` as you know.

But https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/285 is a better way.

Reply:
>> Thanks for sending a fix, but I don't think we could fix this in west.yml at all, could we?
> Well, if we request version `0.7.1` in `west.yml` it would fail with a proper error message if user is using `0.7.0` and with a cryptic message if `0.6.x` as you know.

But we cannot do that in west.yml, because the west version you're referring to (0.7.1) is not the schema version (0.7), and west.yml contains the schema version, not the manifest version.

There is no such thing as schema version 0.7.1. We cannot put 0.7.1 in this line of west.yml.

Reply:
of course, you're absolutely right.
My bad.